plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -458,6 +458,35 @@ func ClusterRoles() []rbac.ClusterRole {
 				eventsRule(),
 			},
 		},
+		{
+			// a role for csi external components {csi-provisioner,csi-attacher,driver-registrar}
+			ObjectMeta: metav1.ObjectMeta{Name: "system:csi-external-provisioner"},
+			Rules: []rbac.PolicyRule{
+				rbac.NewRule("create", "delete", "list", "watch").Groups(legacyGroup).Resources("persistentvolumes").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "update").Groups(legacyGroup).Resources("persistentvolumeclaims").RuleOrDie(),
+				rbac.NewRule("get").Groups(legacyGroup).Resources("secrets").RuleOrDie(),
+				rbac.NewRule("list", "watch").Groups(storageGroup).Resources("storageclasses").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "create", "update", "patch").Groups(legacyGroup).Resources("events").RuleOrDie(),
+			},
+		},
+		{
+			// a role for csi external components {csi-provisioner,csi-attacher,driver-registrar}
+			ObjectMeta: metav1.ObjectMeta{Name: "system:csi-external-attacher"},
+			Rules: []rbac.PolicyRule{
+				rbac.NewRule("get", "list", "watch", "update").Groups(legacyGroup).Resources("persistentvolumes").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch").Groups(legacyGroup).Resources("nodes").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "update").Groups(storageGroup).Resources("volumeattachments").RuleOrDie(),
+				rbac.NewRule("get", "list", "watch", "create", "update", "patch").Groups(legacyGroup).Resources("events").RuleOrDie(),
+			},
+		},
+		{
+			// a role for csi external components {csi-provisioner,csi-attacher,driver-registrar}
+			ObjectMeta: metav1.ObjectMeta{Name: "system:csi-driver-registrar"},
+			Rules: []rbac.PolicyRule{
+				rbac.NewRule("get", "update").Groups(legacyGroup).Resources("nodes").RuleOrDie(),
```

Comment:
I'm not familiar with the registrar component... where does this process run, and what on the Node object is it updating?
/cc @saad-ali

Reply:
The driver registrar adds an annotation on the node that contains a JSON map of driver names to node names. These node names could be different from what Kubernetes views as the node name. The process will generally be deployed alongside the CSI Plugin in a Pod.

Reply:
so this is giving every kubelet access to a credential that can then modify any node in the cluster?

Reply:
I believe so, @saad-ali to confirm